Tests/Admin/Extension/LockExtensionTest.php

```diff
@@ -52,240 +51,155 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
      */
     private $request;
 
-    public function setUp()
+    protected function setUp()
     {
+        $this->modelManager = $this->prophesize('Sonata\AdminBundle\Model\LockInterface');
+        $this->admin = $this->prophesize('Sonata\AdminBundle\Admin\AbstractAdmin');
+
         $this->eventDispatcher = new EventDispatcher();
         $this->request = new Request();
         $this->object = new \stdClass();
         $this->lockExtension = new LockExtension();
-
-        $this->modelManager = $this->getMock('Sonata\AdminBundle\Model\LockInterface');
-
-        $this->admin = $this->getMockBuilder('Sonata\AdminBundle\Admin\AbstractAdmin')
-            ->disableOriginalConstructor()
-            ->getMock();
     }
 
     public function testConfigureFormFields()
     {
         $formMapper = $this->configureFormMapper();
         $form = $this->configureForm();
-        $event = new FormEvent($form, array());
-
-        $this->modelManager->expects($this->once())
-            ->method('getLockVersion')
-            ->will($this->returnValue(1));
-
-        $this->admin->expects($this->once())
-            ->method('getModelManager')
-            ->will($this->returnValue($this->modelManager));
-
-        $form->expects($this->once())
-            ->method('add')
-            ->with(
-                $this->equalTo('_lock_version'),
-                $this->equalTo('hidden'),
-                $this->equalTo(array(
-                    'mapped' => false,
-                    'data' => 1,
-                ))
-            );
+        $this->configureAdmin(null, null, $this->modelManager->reveal());
+        $event = new FormEvent($form->reveal(), array());
 
-        $this->lockExtension->configureFormFields($formMapper);
+        $this->modelManager->getLockVersion(array())->willReturn(1);
```

Comment:
Yes.. I'm not an expert of PHPStorm but since the part where you do:

    $this->modelManager->getLockVersion()

getLockVersion is not a method defined in ObjectProphecy. It is a __call method. Does PHPStorm detect this kind of calls?

Also when you do the `->reveal()` It is returning an `object`, probably extending or implementing the class of your mock but PHPStorm doesn't detect it either.